### PR TITLE
Front-end developer test app submitted by Kiki Ren

### PR DESCRIPTION
**Pull request by kikiren**
- Use css grid approach for Masonry layout
- Responsive column count with media query
- Partial matching filter
- Welcome any feedback :)

### CODE REVIEW

**Comment by flyingant** [comment]
Hi @kikiren , This is MaYi from WCL and reviewing your code.

it looks well on your implementation to the test. 
But I notice that you just loaded all images at once on the home page, would you like to challenge the idea about the lazy loading strategy or just implement it? (Like scroll to load more images?)

Thank you

**Comment by kikiren** [comment]
@flyingant, use intersection observer api to implement lazy loading and infinite scroll, due to the css grid approach, the first picture in each column besides the first column were lazy loaded as they were not intersected then the gallery page loaded, so I changed to a responsive masonry package.

File: src/pages/GalleryPage.js [hunk 1]
```diff
@@ -0,0 +1,35 @@
...
+import Masonry, { ResponsiveMasonry } from "react-responsive-masonry";
+import { setLoadmore } from "redux/actions";
+
+export default function GalleryPage() {
+  const { filter } = useSelector((state) => state);
+  const { images } = useGetImages();
+  const dispatch = useDispatch();
+  useEffect(() => {
+    dispatch(setLoadmore(true));
+  }, []);
```
Comment (johncalvinroberts) [inline src/pages/GalleryPage.js]: I'm actually getting a linting error from this line:
```
src/pages/GalleryPage.js
  Line 14:6: React Hook useEffect has a missing dependency: 'dispatch'. Either include it or remove the dependency array 
```

Do you know why eslint would warn us about this? How might you fix this?

File: src/hooks/useGetImages.js [hunk 2]
```diff
@@ -0,0 +1,27 @@
...
+import { setLoadmore } from "redux/actions";
+
+export default function useGetImages() {
+  const [images, setImages] = useState([]);
+  const { loadMore } = useSelector((state) => state);
+  const dispatch = useDispatch();
+  useEffect(() => {
+    if (loadMore) {
+      // default api
+      let API = "http://localhost:4000/pics";
```
Comment (johncalvinroberts) [inline src/hooks/useGetImages.js]: If you were to use a "real" backend in production, how would you manage this variable?

File: src/hooks/useGetImages.js [hunk 3]
```diff
@@ -0,0 +1,27 @@
...
+      axios
+        .get(API)
+        .then((res) => {
+          setImages([...images, ...res.data]);
+          dispatch(setLoadmore(false));
+        })
+        .catch((e) => {
+          console.log(e);
+        });
```
Comment (johncalvinroberts) [inline src/hooks/useGetImages.js]: Can you update this to use `async/await` syntax?

**Review by johncalvinroberts — COMMENTED** [review]
Hi Kiki. Great job on the frontend test. I left a few comments, feel free to address or change as you see fit.

**Comment by kikiren** [comment]
@johncalvinroberts , 

1. In this case I just want useEffect to behave like componentDidMount , so I added  `// eslint-disable-next-line react-hooks/exhaustive-deps` to disable the eslint rule.
2. I will set env vars for apis as well as secret keys, updated in code.
3. Updated to async/await. :)

**Pull request closed without merging**